static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -72,57 +78,277 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
     this.onCollapseToggle(applicationId)
   }
 
-  renderApplicationDetail = (applicationId: number) => {
-    const {
-      allApplicationDetail,
-      setDrawerOpen,
-      setDrawerState,
-      setDrawerMeta
-    } = this.props
+  /*
+   * Returns an object mapping an application step id to the user's submission (if anything was submitted
+   * for that step).
+   */
+  getAppStepSubmissions = (
+    runApplicationSteps: Array<ApplicationRunStep>,
+    submissions: Array<ApplicationSubmission>
+  ): { [number]: ?ApplicationSubmission } =>
+    R.fromPairs(
+      runApplicationSteps.map((step: ApplicationRunStep) => [
+        step.id,
+        submissions.find(
+          (submission: ApplicationSubmission) =>
+            submission.run_application_step_id === step.id
+        )
+      ])
+    )
 
-    if (!allApplicationDetail || !allApplicationDetail[String(applicationId)]) {
-      return null
-    }
+  /*
+   * Renders a section of the application progress bar (with a checkbox if this step is complete) along with the
+   * detail content of the application step.
+   */
+  renderDetailRow = (
```

Comment:
General feedback that this class is doing too much, each of the `renderX` method should be their own independent components. It'll make the class less complex and easier to test.

Reply:
Yeah, looking at it with fresh eyes this definitely got away from me